### DISCUSSION TITLE
Windows Environment Setup

### DISCUSSION THREAD

**Discussion by scottbisaillon** [typesafegithub/github-workflows-kt#342] [Q&A] [answered]
First of all the project looks great and hits a lot of the points that I did not like about writing Github Actions in yaml so I am very eager to try it out for managing my Flutter app's workflows.

With that being said, I am having a hard time setting up an environment to actually be able to write and run these scripts on my Window's machine. This might be out of the scope of a discussion here, but maybe it will prove useful for future people trying to setup their environments as well.

I do have IntelliJ IDEA installed and setup but how to actually get something working with kotlin scripts is where I am stumped. I don't seem to be able to get certain dependencies to register such that I can use things like `DependsOn` but I might just have some fundamental misunderstandings since I have not worked with kotlin projects very much outside of the minimal setup for a Flutter app.

If anyone has some insight on the direction I should take or any resources I can look at, that would be greatly appreciated. I have searched a bit for the past day or so and haven't had much luck. Thanks!

**Comment by Vampire** [comment, marked as answer]
Are you aware that the file name part `.main.kts` is mandatory?
By having that name suffix, things like `@DependsOn` just work.
To execute it, you can for example install Kotlin and execute it as usual, or you can create an IntelliJ run configuration that executes it, or you can define a Gradle task that executes it.

**Reply by scottbisaillon** [reply]
Okay the `.main.kts` part is something I think I overlooked. Given that information and trying out the example from the docs, I do get a successful execution of the script, but what I am lacking now is just the IDE recognition with syntax highlighting and auto-complete within the file. Is this something I should expect in the context of kotlin scripts?

**Reply by scottbisaillon** [reply]
The issue might have just been transient seeing as invalidating the cache and restarting the IDE seems to have fixed the issue, something I should have known given my experience with the IntelliJ IDEs..

Granted the only I have gotten syntax highlighting and autocompletion to work is making a standalone module within my project which is setup with gradle. Only then I am able to get these features, not when I have my scripts in my `.github/workflow` folder. It could be that I am missing some setup to be able to achieve that (running from that directory works as expected).

**Reply by Vampire** [reply]
No, should just work fine if you have the Kotlin plugin installed. It works fine here directly from the workflows folder and also without a Gradle module or run configuration.

**Reply by scottbisaillon** [reply]
Interesting, might be something weird with my IDEs configuration. I will keep messing around with it. Thanks for the help!

**Reply by krzema12** [reply]
If IntelliJ keeps causing troubles, please attach some screenshots so that it's easier for us to help 😄

**Comment by jmfayard** [comment]
Not using widows, but after installing Kotlin (should be easy to Google) I
would try to execute the Kotlin scripts from our repo/.github/workflows
